WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt

```diff
@@ -0,0 +1,132 @@
+package org.wordpress.android.ui.reader.discover
+
+import dagger.Reusable
+import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.store.AccountStore
+import org.wordpress.android.models.ReaderCardType.DEFAULT
+import org.wordpress.android.models.ReaderCardType.GALLERY
+import org.wordpress.android.models.ReaderCardType.PHOTO
+import org.wordpress.android.models.ReaderCardType.VIDEO
+import org.wordpress.android.models.ReaderPost
+import org.wordpress.android.models.ReaderPostDiscoverData
+import org.wordpress.android.models.ReaderPostDiscoverData.DiscoverType.EDITOR_PICK
+import org.wordpress.android.models.ReaderPostDiscoverData.DiscoverType.OTHER
+import org.wordpress.android.models.ReaderPostDiscoverData.DiscoverType.SITE_PICK
+import org.wordpress.android.ui.reader.discover.ReaderDiscoverViewModel.ReaderCardUiState.ReaderPostUiState
+import org.wordpress.android.ui.reader.discover.ReaderDiscoverViewModel.ReaderCardUiState.ReaderPostUiState.DiscoverLayoutUiState
+import org.wordpress.android.util.DateTimeUtils
+import org.wordpress.android.util.GravatarUtils
+import org.wordpress.android.util.UrlUtils
+import org.wordpress.android.util.image.ImageType.AVATAR
+import org.wordpress.android.util.image.ImageType.BLAVATAR
+import java.lang.IllegalStateException
+import javax.inject.Inject
+
+@Reusable
+class ReaderPostUiStateBuilder @Inject constructor(private val accountStore: AccountStore) {
+    fun mapPostToUiState(post: ReaderPost, photonWidth: Int, photonHeight: Int): ReaderPostUiState {
+        // TODO malinjir onPostContainer click
+        // TODO malinjir on item rendered callback -> handle load more event and trackRailcarRender
+        // TODO malinjir bookmark action
+        // TODO malinjir reblog action
+        // TODO malinjir comments action
+        // TODO malinjir likes action
+
+        return ReaderPostUiState(
+                id = post.postId,
+                blogUrl = buildBlogUrl(post),
+                dateLine = buildDateLine(post),
+                avatarOrBlavatarUrl = buildAvatarOrBlavatarUrl(post),
+                blogName = buildBlogName(post),
+                excerpt = buildExcerpt(post),
+                title = buildTitle(post),
+                photoFrameVisibility = buildPhotoFrameVisbility(post),
```

Comment:
Np: Typo `buildPhotoFrameVisbility` => `buildPhotoFrameVisibility`

Reply:
Fixed in 1d604acb39b501a73ee8bcfdef8a50ab4a8c634a